File: back/src/main/java/csh/back/domain/trip/member/validator/TripMemberValidator.java
package csh.back.domain.trip.member.validator;

import csh.back.domain.trip.member.repository.TripMemberRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service
public class TripMemberValidator {
    private final TripMemberRepository tripMemberRepository;

    public void validMember(Long tripGroupId, Long memberId) {
        if(!tripMemberRepository.existsByTripGroupIdAndMemberId(tripGroupId, memberId)) {
            throw new RuntimeException();
        }
    }
}

File: back/src/main/kotlin/csh/back/domain/trip/member/validator/TripMemberValidator.kt
package csh.back.domain.trip.member.validator

import csh.back.domain.trip.member.repository.TripMemberRepository
import org.springframework.stereotype.Service

@Service
class TripMemberValidator(
    private val tripMemberRepository: TripMemberRepository,
) {
    fun validMember(tripId: Long, memberId: Long) {
        if (!tripMemberRepository.existsByTripGroupIdAndMemberId(tripId, memberId)) {
            throw RuntimeException()
        }
    }
}
